--- splashutils-1.5.4.3/miscsplashutils/fbtruetype/Makefile~	2012-02-12 11:01:53.693422419 +0100
+++ splashutils-1.5.4.3/miscsplashutils/fbtruetype/Makefile	2012-02-12 11:02:43.013691625 +0100
@@ -23,8 +23,8 @@
 OBJECTS  = fbtruetype.o messages.o console.o ttf.o luxisri.o
 
 LDFLAGS  =  -L$(LIBDIR)
-LIBSS    = $(LIBDIR)/libfreetype.a $(LIBDIR)/libz.a -lm
-LIBSD    = -lfreetype -lm
+LIBSS    = $(LIBDIR)/libfreetype.a $(LIBDIR)/libz.a $(LIBDIR)/libbz2.a -lm
+LIBSD    = -lfreetype -lm -lbz2
 
 CFLAGS   = -O2 -Wall -I/usr/include/freetype2
 COMPILE  = $(CC) $(CFLAGS)
