--- splashutils-1.5.4/Makefile.am~	2008-02-25 21:41:35.000000000 +0200
+++ splashutils-1.5.4/Makefile.am	2008-04-16 02:30:05.999841930 +0300
@@ -1,4 +1,4 @@
-SUBDIRS = libs src docs misc scripts
+SUBDIRS = src docs misc scripts
 
 include common.mk
 
--- splashutils-1.5.4/src/Makefile.am	2008-04-16 03:02:06.179841730 +0300
+++ splashutils-1.5.4/src/Makefile.am	2008-04-16 16:55:34.161432745 +0300
@@ -236,40 +236,15 @@
-fbcondecor_helper_CFLAGS   += -I$(abs_top_builddir)/libs/libjpeg -I$(LIBJPEG_SOURCE)
-fbcondecor_helper_LDADD    += $(top_builddir)/libs/libjpeg.a
+fbcondecor_helper_LDADD    += -ljpeg
 if CONFIG_PNG
-fbcondecor_helper_CFLAGS   += -I$(LIBPNG_SOURCE)
-fbcondecor_helper_LDADD    += $(top_builddir)/libs/libpng.a $(top_builddir)/libs/libz.a
+fbcondecor_helper_LDADD    += -lpng -lz
 endif
 if CONFIG_TTF_KERNEL
 fbcondecor_helper_SOURCES  += ttf.c ttf.h fbsplash.h
-fbcondecor_helper_CFLAGS   += -I$(LIBFREETYPE2_SOURCE)/include -I$(LIBZ_SOURCE)
-fbcondecor_helper_LDADD    += $(top_builddir)/libs/libfreetype.a
+fbcondecor_helper_CFLAGS   += -I/usr/include/freetype2
+fbcondecor_helper_CFLAGS   += -I/usr/include
+fbcondecor_helper_LDADD    += -lfreetype
 endif
 if CONFIG_KLIBC_SHARED
 fbcondecor_helper_LDFLAGS  += -shared
 else
 fbcondecor_helper_LDFLAGS  += -static
 endif
-
-# Override automake's generation of target rules for building fbcondecor_helper.
-fbcondecor_helper: $(fbcondecor_helper_LDADD) $(fbcondecor_helper_OBJECTS)
-	@$(call infmsg,LD,$@)
-	$(Q)$(MKDIR_P) $(@D)
-	$(Q)$(KLCC) $(fbcondecor_helper_CFLAGS) $(fbcondecor_helper_LDFLAGS) $+ $(fbcondecor_helper_LDADD) -o $@
-	$(Q)$(STRIP) $@
-
-# Override automake's generation of target rules for building fbcondecor_helper object files.
-fbcondecor_helper-kernel.o:
-fbcondecor_helper-libfbsplash.o:
-fbcondecor_helper-libfbsplashrender.o:
-fbcondecor_helper-fbcon_decor.o:
-fbcondecor_helper-common.o:
-fbcondecor_helper-parse.o:
-fbcondecor_helper-list.o:
-fbcondecor_helper-render.o:
-fbcondecor_helper-image.o:
-fbcondecor_helper-effects.o:
-fbcondecor_helper-ttf.o:
-fbcondecor_helper-%.o: %.c
-	@$(call infmsg,CC,$@)
-	$(Q)$(MKDIR_P) $(@D)
-	$(Q)$(KLCC) $(fbcondecor_helper_CPPFLAGS) $(fbcondecor_helper_CFLAGS) $(INCLUDES) -c $< -o $@
--- splashutils-1.5.4.3/miscsplashutils/fbtruetype/Makefile~	2006-02-03 16:10:41.000000000 +0100
+++ splashutils-1.5.4.3/miscsplashutils/fbtruetype/Makefile	2012-01-31 21:43:52.329619142 +0100
@@ -23,7 +23,7 @@
 OBJECTS  = fbtruetype.o messages.o console.o ttf.o luxisri.o
 
 LDFLAGS  =  -L$(LIBDIR)
-LIBSS    = $(LIBDIR)/libfreetype.a $(LIBDIR)/libz.a -lm
+LIBSS    = $(LIBDIR)/libfreetype.a $(LIBDIR)/libz.a $(LIBDIR)/libbz2.a -lm
 LIBSD    = -lfreetype -lm
 
 CFLAGS   = -O2 -Wall -I/usr/include/freetype2
